### PR TITLE
Issue #133

### PR DESCRIPTION
**Pull request by Lomet**
add code
need fix test

### CODE REVIEW

File: contracts/DelayVault.sol [hunk 1]
```diff
@@ -53,7 +53,7 @@ contract DelayVault is DelayView {
         vault.StartDelay = _startDelay;
         vault.CliffDelay = _cliffDelay;
         vault.FinishDelay = _finishDelay;
-        Array.addIfNotExsist(Users[_token], msg.sender);
+        Array.addIfNotExsist(TokenToUsers[_token], msg.sender);
```
Comment (Lomet) [inline contracts/DelayVault.sol:56]: same

File: contracts/DelayEvents.sol [hunk 2]
```diff
@@ -18,7 +18,12 @@ contract DelayEvents {
         uint256[] CliffDelays,
         uint256[] FinishDelays
     );
-    event BoughtBackTokens(
+    event TokenRedemptionApproval(
+        address indexed token,
```
Comment (YouStillAlive) [inline contracts/DelayEvents.sol:22]: Events use different conversation names. Change `token` to `Token`, etc.

File: contracts/DelayEvents.sol [hunk 3]
```diff
@@ -18,7 +18,12 @@ contract DelayEvents {
...
+        address indexed user,
+        bool status
+    );
+    event RedeemedTokens(
```
Comment (YouStillAlive) [inline contracts/DelayEvents.sol:26]: What do you think about adding the "Owner" field to the RedeemedTokens event?

**Review by YouStillAlive — CHANGES_REQUESTED** [review]

File: contracts/DelayEvents.sol (hunk 3 above)
Comment (Lomet) [reply contracts/DelayEvents.sol:26]: I don't see a reason. only if we add address to the redemption, then we want to know who get the tokens

**Review by YouStillAlive — APPROVED** [review]

**Pull request merged**